src/common/snippets/docs/mha_optimization_guide.md

```diff
@@ -0,0 +1,150 @@
+
+## MHA Tokenization
+
+This structure represents the basic MHA pattern that can be tokenized by Snippets:
+
+```mermaid
+graph TB
+    MM0A[Transpose] --> MatMul0
+    MM0B[Transpose/Eltwise/FakeQuantize] --> MatMul0
+    MatMul0 --> IntermediateBeforeSM[Transpose/Eltwise/Select/Reshape/FakeQuantize]
+    IntermediateBeforeSM --> Softmax
+    Softmax --> IntermediateAfterSM[Transpose/Eltwise/Select/Reshape/FakeQuantize]
+    IntermediateAfterSM --> MatMul1
+    MM1B[Transpose] --> MatMul1
+    MatMul1 --> OpAfterMM2[Transpose/Eltwise/FakeQuantize]
+```
+
+The main layers in MHA pattern are `MatMul0`, `Softmax` and `MatMul1`. Other layers are optional.
+Please note, that layers, denoted by `/`, can represent both single nodes and sequences of nodes.
+The code, which performs the tokenization, is placed in [TokenizeMHASnippets](../src/pass/mha_tokenization.cpp) transformation.
+
+### CPU Plugin Callback for MHA Tokenization
+
+The tokenization pass can be adjusted via callback
+In CPU plugin, the callback disables tokenization in 3 types of cases:
+
+1. Operations that are not supported by Snippets CPU backend.
+For example, because fusing is not expected to bring sufficient optimization opportunities.
+2. Operations skipped deliberately to allow for plugin-specific fusings.
+For example, elementwise operations that follow Convolution nodes are skipped because the eltwises will be fused into Convolutions by the CPU plugin.
+3. Operations that are not tokenized for performance reasons: executing MHA operations one-by-one may be faster in some cases.
+
+The CPU plugin callback for TokenizeMHASnippets is placed in [transformation_pipeline.cpp](../../../plugins/intel_cpu/src/transformations/transformation_pipeline.cpp) file (please see the code in `MainSnippets` method).
+
+**Please note that the CPU callback is usually ignored in cpu functional tests: SnippetsMode::IgnoreCallback is used for that**.
+Currently, SnippetsMode has 3 states: `Enable`, `IgnoreCallback` and `Disable`.
+For the details, please refer to [ov::intel_cpu::Config](../../../plugins/intel_cpu/src/config.h).
+
+## Snippets Common Optimizations
+
+After tokenization, snippets [common optimizations](../src/pass/common_optimizations.cpp) are applied to the tokenized Subgraphs.
+These transformations can modify both the Subgraph's body and its surroundings (e.g. extract constant nodes outside the Subgraph).
+Let's explore several transformations that can impact MHA performance.
+
+### ExtractUnsupportedTransposes
+
+[ExtractUnsupportedTransposes](../src/pass/extract_unsupported_transposes.cpp) moves up unsupported Transposes outside the Subgraph.
+
+Snippets support 2 types of Transposes:
+
+1. Transposes on Matmul inputs and output: these Transposes will be fused into Brgemm (which supports strided read/write) node by [FuseTransposeBrgemm](../src/pass/fuse_transpose_brgemm.cpp) data flow transformation.
```

Comment:
According to the diagram you provided, all Transposes are on Matmult's inputs or outputs, so this distinction is a bit confusing.
I think the main criteria, that that the transpose doesn't change the last dimension => can be executed inplace by specifying strides. It's already mentioned, let's just remove the first part before `:` and repharase a little

Reply:
Done, thanks!